compat/krb5.h

```diff
@@ -46,4 +46,9 @@
 #  endif
 #endif /* HAVE_KRB5_H */
 
+#if !HAVE_KRB5_GET_MAX_TIME_SKEW && HAVE_MAX_SKEW_IN_KRB5_CONTEXT
+#define krb5_get_max_time_skew(x) (x)->max_skew
+#define HAVE_KRB5_GET_MAX_TIME_SKEW 1
+#endif
```

Comment:
I think this should be refactored to avoid redefining `HAVE_KRB5_GET_MAX_TIME_SKEW` and to avoid defining a macro with the standard library name. Here is an untested sketch:

```suggestion
/// krb5_get_max_time_skew() portability wrapper
/// \retval DEFAULT_SKEW if we do not know how to get the maximum
krb5_deltat
GetMaxTimeSkew(const krb5_context &context)
{
#if HAVE_KRB5_GET_MAX_TIME_SKEW
    return krb5_get_max_time_skew(&context);
#elif HAVE_MAX_SKEW_IN_KRB5_CONTEXT
    return context.max_skew;
#else
    (void)context;
    return DEFAULT_SKEW;
#endif
}
```

Please adjust as needed. The caller will need to be trivially adjusted as well, of course.

Alternatively, this simple logic should be moved into the caller itself (until it needs to be _reused_ elsewhere). 

The above GetMaxTimeSkew() suggestion makes the logic reusable but may require more work. I am OK with either approach.

Reply:
Absolutely Not.
`krb5_get_max_time_skew(x)` is the official library API we are using in the main Squid code. This libcompat portability code is **only** providing a replacement hack for that API call as-needed when the distro build environment library does not properly present it (eg outdated library version).

Reply:
AFAICT, the facts in your response do not contradict and do not address my change request.

Reply:
Skew hack has been dropped from recent Squid code.